examples/callback/example_callback.cpp

```diff
@@ -135,14 +135,12 @@ int main(int argc, char *argv[])
 
   {
     // EXPERIMENTAL
-    // TODO replace with BVH::query(ExecutionSpace, Predicates, Callback) when
-    // new overload is added
     Kokkos::View<int, ExecutionSpace, Kokkos::MemoryTraits<Kokkos::Atomic>> c(
         "counter");
 
-    ArborX::Details::traverse(
-        ExecutionSpace{}, bvh, FirstOctant{},
-        KOKKOS_LAMBDA(int i, int j) { printf("%d %d %d\n", ++c(), i, j); });
+    bvh.query(ExecutionSpace{}, FirstOctant{}, KOKKOS_LAMBDA(int i, int j) {
+      printf("%d %d %d\n", ++c(), i, j);
```

Comment:
To be a valid example, and to indicate to users that there is no guarantee on threading, need to increment `c` atomically.

Reply:
Did you look how I declared the view?

Reply:
Oh, yeah, I now remember this in the previous PR, forgot when looking at this diff.